.github/workflows/ci.yaml

```diff
@@ -12,7 +12,7 @@ jobs:
       - name: Setup Go
         uses: actions/setup-go@v3
         with:
-          go-version: 1.19
+          go-version: 1.20.10
```

Comment:
Why use a point release and not just `1.20`?

Reply:
Fixed

Reply:
my question was why use 1.XX.YY instead of just 1.XX? then we won't have to update for every point release

Reply:
> my question was why use 1.XX.YY instead of just 1.XX? then we won't have to update for every point release

@micahhausler the YY bump could mean some CVE fixes in Golang minor version, thus we can follow to patch the component and cut a new image tag. Otherwise, there is no change in the component and we can't cut a new image tag.

Reply:
@micahhausler not sure why, but 1.20 in the github workflow has failure in CI build job: https://github.com/kubernetes-sigs/aws-encryption-provider/actions/runs/6873400343, 1.20.10 is working fine

Reply:
We do this in kubernetes so it's still a controlled PR to upgrade and builds are more reproducible (this isn't itself enough to get reproducible builds however) but YMMV.

Reply:
@BenTheElder With this change to adopt gimme, everything is predictable, the GitHub workflow Go version actually doesn't matter any more, in both CI and local environment, the build system will choose the Go version in .go-image-tag as the Go version for building binary, running tests and building image (currently there is a small gap for me to adopt .go-version).

I'm more down to 1.XX.YY Go version in the GitHub workflow than the 1.XX, since it is more predictable and we want to follow Kubernetes to update go version in the project to have more predictable tests & build and better security posture.

since with gimme it doesn't matter anymore, actually we should be able to get rid of it.

What's your thoughts @micahhausler ?